secret.go

```diff
@@ -123,24 +149,89 @@ func getGitSecret(userVal string) string {
 	}
 	return gitPat
 }
-func isFileTypeSecret(secType string) bool {
-	if secType == "gcp" {
+func isFileTypeSecret(secretType string) bool {
+	switch {
+	case strings.EqualFold(secretType, GCP):
+		return true
+	case strings.EqualFold(secretType, SSHKey):
 		return true
+	default:
+		return false
 	}
-	return false
+
 }
-func createSecret(secretName string, identifier string, secretValue string, secretType string) HarnessSecret {
+func createSecret(secretName string, identifier string, secretValue string, secretType string, secretData SSHWINRMSecretData) HarnessSecret {
 	typeOfSecret := "SecretText"
+
+	var newSecret HarnessSecret
 	if secretType != "" {
 		typeOfSecret = secretType
 	}
 	if identifier == "" {
 		identifier = strings.ReplaceAll(secretName, " ", "_")
 	}
-	newSecret := HarnessSecret{Secret: Secret{Type: typeOfSecret, Name: secretName, Identifier: identifier, ProjectIdentifier: DEFAULT_PROJECT,
-		OrgIdentifier: DEFAULT_ORG, Spec: SecretSpec{SecretManagerIdentifier: "harnessSecretManager", ValueType: "Inline"}}}
-	if secretType == SecretText {
-		newSecret.Spec.Value = secretValue
+
+	if strings.EqualFold(secretType, SSHKey) {
+
+		secretTypeData := SSHSecretType{
+			Auth: SecretAuth{
+				Type: SShSecretType,
+				Spec: SSHSecretSpec{
+					CredentialType: "KeyReference",
+					Spec: SShSecretSubSpec{
+						UserName: "",
+						Key:      "",
+					},
+				},
+			},
+			Port: 22,
+		}
+		newSecret = HarnessSecret{Secret: Secret{Type: typeOfSecret, Name: secretName, Identifier: identifier, ProjectIdentifier: DEFAULT_PROJECT,
+			OrgIdentifier: DEFAULT_ORG, Spec: secretTypeData,
+		}}
+
+		if sshspec, ok := newSecret.Spec.(SSHSecretType); ok {
+			sshspec.Port = secretData.Port
+			sshspec.Auth.Spec.Spec.UserName = secretData.Username
+			sshspec.Auth.Spec.Spec.Key = secretData.Key
+
+			newSecret.Spec = sshspec
+		}
+	} else if strings.EqualFold(secretType, WinRM) {
+
+		secretTypeData := WinRMSecretType{
+			Auth: WinRMSecretAuth{
+				Type: NTLM,
+				Spec: WinRMSecretSpec{
+					Username: "",
+					Password: "",
+					Domain:   "",
+				},
+			},
+			Port:       5985,
```

Comment:
Let's not hardcode the Ports here and define them in the constants file with a descriptive name ?

Reply:
this is just a default value we are already taking it as input

Reply:
think of it as placeholder